sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -282,17 +282,21 @@ class GpuHashAggregateIterator(
   private def aggregateInputBatches(): Unit = {
     val aggHelper = new AggHelper(forceMerge = false, useTieredProject = useTieredProject)
     while (cbIter.hasNext) {
-      withResource(cbIter.next()) { childBatch =>
-        val isLastInputBatch = GpuColumnVector.isTaggedAsFinalBatch(childBatch)
-        withResource(computeAggregate(childBatch, aggHelper)) { aggBatch =>
-          val batch = LazySpillableColumnarBatch(aggBatch, metrics.spillCallback, "aggbatch")
-          // Avoid making batch spillable for the common case of the last and only batch
-          if (!(isLastInputBatch && aggregatedBatches.isEmpty)) {
-            batch.allowSpilling()
-          }
-          aggregatedBatches.add(batch)
+      val (aggBatch, isLastInputBatch) =
+        closeOnExcept(cbIter.next()) { childBatch =>
```

Comment:
I don't think we want this here, as `computeAggregateAndClose` is responsible for closing the batch, whether an exception occurs or not, and if one does occur, this will try to close it again and get a double-close error.